packages/rpc-graphql/src/loaders/account.ts

```diff
@@ -1,45 +1,334 @@
+            return {
+                ...account,
+                data: slicedData(data, 'base58'),
+            };
+        }
+    }
+    return account;
+}
+
+/**
+ * Load an account using the RPC's `getAccountInfo` method.
+ */
+async function loadAccount(rpc: Rpc<GetAccountInfoApi>, { address, ...config }: AccountLoaderArgs) {
     return await rpc
         .getAccountInfo(address, config)
         .send()
         .then(res => res.value);
 }
 
-function createAccountBatchLoadFn(rpc: Rpc<GetAccountInfoApi>) {
+/**
+ * Load multiple accounts using the RPC's `getMultipleAccounts` method.
+ */
+async function loadMultipleAccounts(
+    rpc: Rpc<GetMultipleAccountsApi>,
+    { addresses, ...config }: MultipleAccountsLoaderArgs,
+) {
+    return await rpc
+        .getMultipleAccounts(addresses, config)
+        .send()
+        .then(res => res.value);
+}
+
+const hashOmit = (args: AccountLoaderArgsBase, omit: (keyof AccountLoaderArgsBase)[]) => {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const argsObj: any = {};
+    for (const [key, value] of Object.entries(args)) {
+        if (!omit.includes(key as keyof AccountLoaderArgsBase)) {
+            argsObj[key] = value;
+        }
+    }
+    return cacheKeyFn(argsObj);
+};
+
+function createAccountBatchLoadFn(rpc: Rpc<GetAccountInfoApi & GetMultipleAccountsApi>) {
     const resolveAccountUsingRpc = loadAccount.bind(null, rpc);
-    return async (accountQueryArgs: readonly AccountLoaderArgs[]) =>
-        accountQueryArgs.map(async args => await resolveAccountUsingRpc(applyDefaultArgs(args)));
+    const resolveMultipleAccountsUsingRpc = loadMultipleAccounts.bind(null, rpc);
+    return async (accountQueryArgs: readonly AccountLoaderArgs[]): Promise<Promise<AccountLoaderValue>[]> => {
+        /**
+         * Gather all the accounts that need to be fetched, grouped by address.
+         */
+        const accountsToFetch: {
+            [address: string]: Readonly<{
+                args: AccountLoaderArgsBase;
+                promiseCallback: AccountBatchLoadPromiseCallback;
+            }>[];
+        } = {};
+        try {
+            return accountQueryArgs.map(
+                ({ address, ...args }) =>
+                    new Promise((resolve, reject) => {
+                        const accountRecords = (accountsToFetch[address] ||= []);
+                        accountRecords.push({ args, promiseCallback: { reject, resolve } });
+                    }),
+            );
+        } finally {
+            /**
+             * Maximum number of acceptable bytes to waste before splitting two
+             * `dataSlice` requests into two requests.
+             */
+            const maxByteRange = 200;
+
+            /**
+             * Maximum number of accounts to fetch in a single batch.
+             * See https://docs.solana.com/api/http#getmultipleaccounts.
+             */
+            const maxMultipleAccountsBatchSize = 100;
```

Comment:
Should probably make these into arguments that bubble up all the way up to `createGraphQLRpc()`, so that if you have an RPC URL that imposes a _lower_ `gMA` limit, or one with tolerance for a higher byte window, the RPC-bringer can configure that.

Reply:
Done!